hive-metastore/src/main/java/org/apache/iceberg/hive/CachedClientPool.java

```diff
@@ -56,9 +57,16 @@ public class CachedClientPool implements ClientPool<IMetaStoreClient, TException
     init();
   }
 
+  @VisibleForTesting
+  static String cacheKey(Configuration conf) {
+    return String.format(
+        "%s:%s",
+        conf.get(HiveConf.ConfVars.METASTOREURIS.varname, ""), conf.get(CATALOG_DEFAULT, ""));
```

Comment:
If I understand correctly Spark sends every configuration value to the HMS Client which starts with `spark.sql.catalog.catalog1.hadoop.` in this case.
Choosing `metastore.catalog.default` as the only thing used in the cacheKey seems like an ad-hoc decision to me. Some interesting configs:
- CLIENT_SOCKET_TIMEOUT
- HMS_HANDLER_ATTEMPTS
- HMS_HANDLER_INTERVAL
Different users might require different values here...

Would it be "overengineering" to have a static list of config keys to use in the `cacheKey`, and use a hash, or a join of the values for the cache key?

Reply:
Yea I was mentioning this too, and internally debating whether its overengineering.  But I wonder , is the list of config something we should expose say via HiveCatalog or just a static list?   I also dont like that if we change the timeout, it can take 5 minutes for a new client to spawn with that.

Reply:
> Would it be "overengineering" to have a static list of config keys to use in the cacheKey, and use a hash, or a join of the values for the cache key?

That's a good question. The use case of `metastore.catalog.default` is reasonable since users move data between them, so they have to be in the same job, and to be distinguished. I'd assume it is rare to configure 2 spark catalogs within **one job** for configurations like, CLIENT_SOCKET_TIMEOUT, HMS_HANDLER_ATTEMPTS, HMS_HANDLER_INTERVAL. But I'm open to suggestion.

Reply:
Yea for Spark job it may be rare but HiveCatalog could be used beyond it, like in a long lived server, for example.

Reply:
I do not know of a situation where someone is currently restricted by this configuration problem, but I feel that the static list would be a small additional complexity which would show a clear path forward for anyone bumping into the situation later, even if we are not here to review it.

Alternatively:
- New configuration: `cacheKey`
- If the `cacheKey` is provided use it for cache
- If the `cacheKey` is not provided then use the `uri`

This would solve all of our problems and it would provide the maximum flexibility for the user. The question is security, but I think that could be solved as well.